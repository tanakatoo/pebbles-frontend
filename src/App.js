import './styles/app.css';
import NavBar from "./components/common/NavBar"
import "./styles/app.css"
import { useSelector } from 'react-redux';
import MyRoutes from "./MyRoutes"

function App() {
  //get the font from the store
  const font = useSelector(state => state.langFont.font)

  //we have to hardcode the string of the font because tailwind doesn't know what to use at build
  return (
    <div className={`App ${font == 'font-poppins' ? 'font-poppins' : 'font-notoJP'}`}>
      <NavBar />
      <MyRoutes />
    </div >
  );
}

export default App;
