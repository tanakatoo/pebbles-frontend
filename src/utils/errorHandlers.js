import { rest } from 'msw'

let loginResponse = {
    token: 'abcdefg',
    user: {
        id: 2,
        username: 'hello',
        role: 'regular',
        premium_join_date: '',
        premium_end_date: '',
        free_trial_start_date: '',
    }
}



export const errorHandlers = [

    //login
    rest.post('http://localhost:3001/auth/login', (req, res, ctx) => {

        console.log(res(ctx.json(['NOT_FOUND'])))
        return res(ctx.json(['NOT_FOUND']));
    }),

]