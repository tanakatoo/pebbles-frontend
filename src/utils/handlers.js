import { rest } from 'msw'

let loginResponse = {
    token: 'abcdefg',
    user: {
        id: 2,
        username: 'hello',
        role: 'regular',
        premium_join_date: '',
        premium_end_date: '',
        free_trial_start_date: '',
    }
}



export const handlers = [

    //login
    rest.post('http://localhost:3001/auth/login', (req, res, ctx) => {

        console.log(res(ctx.json(loginResponse)))
        return res(ctx.json(loginResponse));
    }),

    rest.post('http://localhost:3001/auth/logout', (req, res, ctx) => {

        console.log('MOCK /logout')
        window.localStorage.setItem('token', null)
        return res(ctx.json());
    }),


    // Called via <ExploreCommunity type="studyBuddy" />
    rest.get('http://localhost:3001/study-buddies/1', (req, res, ctx) => {
        console.log('MOCK /study-buddies/1');

        return res(ctx.json([]));
    }),

    // Called via <ExploreCommunity type="studyBuddy" />
    rest.post('http://localhost:3001/auth/register', (req, res, ctx) => {
        console.log('MOCK /auth/register');

        return res(ctx.json(loginResponse));
    }),

]