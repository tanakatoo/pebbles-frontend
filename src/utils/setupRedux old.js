
import { configureStore } from '@reduxjs/toolkit';
import rootReducer from "../reducers/rootReducer"
import { persistStore, persistReducer, FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER } from 'redux-persist';

import storage from 'redux-persist/lib/storage'
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'

export const persistConfig = {
    key: 'root', // Key prefix for the persisted state
    storage, // Storage mechanism (e.g., local storage)
    stateReconciler: autoMergeLevel2, // Specify which parts of the state to persist
    whitelist: ['profile', 'langFont']
};

export const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = configureStore({
    reducer: persistedReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            serializableCheck: {
                ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
            }
        })
})
export const persistedStore = persistStore(store)