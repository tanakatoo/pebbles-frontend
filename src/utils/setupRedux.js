import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
// As a basic setup, import your same slice reducers
import { persistStore, persistReducer, FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER } from 'redux-persist';
import { PersistGate } from 'redux-persist/integration/react'
import { ListSlice } from "../reducers/rootReducer";
import rootReducer from "../reducers/rootReducer"
import storage from 'redux-persist/lib/storage'
import storeSetup from "./storeSetup"
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'

// const persistConfig = {
//     key: 'root', // Key prefix for the persisted state
//     storage, // Storage mechanism (e.g., local storage)
//     stateReconciler: autoMergeLevel2, // Specify which parts of the state to persist
//     whitelist: ['profile', 'langFont']
// };
// const persistedReducer = persistReducer(persistConfig, rootReducer);

// const persistedStore = persistStore(store)
// const root = ReactDOM.createRoot(document.getElementById('root'));


export function renderWithProviders(
    ui,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = storeSetup(preloadedState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {

        return <Provider store={store}>{children}</Provider>;
    }

    // Return an object with the store and all of RTL's query functions
    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}