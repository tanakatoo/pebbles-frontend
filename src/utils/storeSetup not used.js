import { combineReducers, configureStore } from '@reduxjs/toolkit'

import rootReducer from '../reducers/rootReducer'
// import userReducer from '../features/users/userSlice'

// Create the root reducer separately so we can extract the RootState type
// const rootReducer = combineReducers({
//     user: userReducer
// })

export const setupStore = preloadedState => {
    return configureStore({
        reducer: rootReducer,
        preloadedState
    })
}