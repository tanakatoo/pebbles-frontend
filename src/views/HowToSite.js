import React, { useEffect } from 'react'
import usePageText from '../hooks/usePageText'
import { HashLink } from 'react-router-hash-link'
import { AwesomeArrowRightLong, AwesomeArrowUpLong } from '../styles/Icons'

function HowToSite() {
    const [pageText, lang] = usePageText("howToSite")
    useEffect(() => {
        window.scrollTo(0, 0)
    }, [])

    return (
        <div className='py-8 flex flex-col border-t-2 border-gray' id="top">
            <h1 className='text-center font-bold text-mobile-header-1 mb-8'>{pageText.TITLE}</h1>
            <div className='w-full flex flex-col pt-4 md:pt-8 pb-12 px-2 bg-primary-super-light '>
                <div className=' mx-auto flex flex-col '>

                    <div className='my-1 ' >
                        <div className='group'>
                            <HashLink to="#link-profile" className=' group-hover:underline group-hover:underline-offset-4 group-hover:text-gray-text'>
                                {pageText.PROFILE}
                                <span className='inline-block group-hover:translate-x-2 group-hover:text-gray-text transition text-black ps-1'>
                                    <AwesomeArrowRightLong />
                                </span>
                            </HashLink>
                        </div>
                    </div>
                    <div className='my-1 '>
                        <div className='group'>
                            <HashLink to="#link-studyBuddy" className=' group-hover:underline group-hover:underline-offset-4 group-hover:text-gray-text'>
                                {pageText.STUDY_BUDDY}
                                <span className='inline-block group-hover:translate-x-2 group-hover:text-gray-text transition text-black ps-1'>
                                    <AwesomeArrowRightLong />
                                </span>
                            </HashLink>
                        </div>
                    </div>
                    <div className='my-1 '>
                        <div className='group'>
                            <HashLink to="#link-studyBuddyUse" className=' group-hover:underline group-hover:underline-offset-4 group-hover:text-gray-text'>
                                {pageText.STUDY_BUDDY_USE}
                                <span className='inline-block group-hover:translate-x-2 group-hover:text-gray-text transition text-black ps-1'>
                                    <AwesomeArrowRightLong />
                                </span>
                            </HashLink>
                        </div>
                    </div>

                </div>
            </div >

            <div className='container flex flex-col mx-auto mt-8'>
                <div className='max-w-prose mx-auto'>
                    <h2 id="link-profile" className='me-auto text-mobile-section-header font-medium my-4'>
                        {pageText.PROFILE}</h2>
                    <p>{pageText.PROFILE_BODY}</p>
                </div>
                <div className='my-4 flex w-full justify-end' >
                    <HashLink to="#top">
                        <AwesomeArrowUpLong />
                        <span className='text-mobile-label-1 ms-1'>{pageText.TOP}</span></HashLink></div>

                <div className='max-w-prose mx-auto'>
                    <h2 id="link-studyBuddy" className='pt-4 max-w-prose me-auto text-mobile-section-header font-medium my-4'>
                        {pageText.STUDY_BUDDY}</h2>
                    <p>{pageText.STUDY_BUDDY_BODY}</p>
                </div>
                <div className='my-4 flex w-full justify-end' >
                    <HashLink to="#top">
                        <AwesomeArrowUpLong />
                        <span className='text-mobile-label-1 ms-1'>{pageText.TOP}</span></HashLink></div>

                <div className='max-w-prose mx-auto'>
                    <h2 id="link-studyBuddyUse" className='pt-4 max-w-prose me-auto text-mobile-section-header font-medium my-4'>
                        {pageText.STUDY_BUDDY_USE}</h2>
                    <p>{pageText.STUDY_BUDDY_USE_BODY}</p>
                </div>
                <div className='my-4 flex w-full justify-end' >
                    <HashLink to="#top">
                        <AwesomeArrowUpLong />
                        <span className='text-mobile-label-1 ms-1'>{pageText.TOP}</span></HashLink></div>
            </div>
        </div>
    )
}
export default HowToSite