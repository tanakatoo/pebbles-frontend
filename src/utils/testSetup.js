// import { persistedStore, store } from './setupRedux';
import { render } from '@testing-library/react'
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { setupTestStore } from './setupRedux';


const renderWithProviders = (ui, options = {}) => {
    const [store, persistedStore] = setupTestStore(options.preloadedState);

    const AllProviders = ({ children }) => {
        return (
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistedStore}>
                    {children}
                </PersistGate>
            </Provider>
        )
    };

    return { store, ...render(ui, { wrapper: AllProviders, ...options }) };
}


// "re-export everything"
export * from '@testing-library/react';

export { renderWithProviders };