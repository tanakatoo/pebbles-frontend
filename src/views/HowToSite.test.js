import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { renderWithProviders } from '../utils/testSetup';
import HowToSite from './HowToSite';
import App from '../App';


// Mock window.scrollTo()
window.scrollTo = jest.fn();
afterAll(() => {
    jest.clearAllMocks();
});

beforeEach(async () => {
    window.localStorage.clear()

})

afterEach(() => {
    window.localStorage.clear()
})


describe('HowToSite', () => {
    test('renders without crashing', () => {
        renderWithProviders(
            <MemoryRouter initialEntries={["/faq"]}>
                <App />
            </MemoryRouter>
        );
    });

    test('matches snapshot', async () => {
        const { asFragment } = renderWithProviders(
            <MemoryRouter initialEntries={["/faq"]}>
                <App />
            </MemoryRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});