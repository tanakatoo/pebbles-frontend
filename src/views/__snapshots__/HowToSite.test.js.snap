// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowToSite matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="py-8 flex flex-col border-t-2 border-gray"
    id="top"
  >
    <h1
      class="text-center font-bold text-mobile-header-1 mb-8"
    >
      Learn Pebbles
    </h1>
    <div
      class="w-full flex flex-col pt-4 md:pt-8 pb-12 px-2 bg-primary-super-light "
    >
      <div
        class=" mx-auto flex flex-col "
      >
        <div
          class="my-1 "
        >
          <div
            class="group"
          >
            <a
              class=" group-hover:underline group-hover:underline-offset-4 group-hover:text-gray-text"
              href="/#link-profile"
            >
              Set up your profile
              <span
                class="inline-block group-hover:translate-x-2 group-hover:text-gray-text transition text-black ps-1"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-arrow-right-long fa-1x "
                  data-icon="arrow-right-long"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </a>
          </div>
        </div>
        <div
          class="my-1 "
        >
          <div
            class="group"
          >
            <a
              class=" group-hover:underline group-hover:underline-offset-4 group-hover:text-gray-text"
              href="/#link-studyBuddy"
            >
              Becoming and delisting from study buddy
              <span
                class="inline-block group-hover:translate-x-2 group-hover:text-gray-text transition text-black ps-1"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-arrow-right-long fa-1x "
                  data-icon="arrow-right-long"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </a>
          </div>
        </div>
        <div
          class="my-1 "
        >
          <div
            class="group"
          >
            <a
              class=" group-hover:underline group-hover:underline-offset-4 group-hover:text-gray-text"
              href="/#link-studyBuddyUse"
            >
              Using the Study Buddy Board
              <span
                class="inline-block group-hover:translate-x-2 group-hover:text-gray-text transition text-black ps-1"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-arrow-right-long fa-1x "
                  data-icon="arrow-right-long"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container flex flex-col mx-auto"
    >
      <h2
        class="max-w-prose me-auto text-mobile-section-header font-medium my-4"
        id="link-profile"
      >
        Set up your profile
      </h2>
      <p
        class="max-w-prose mx-auto"
      >
        Once you login, click on the avatar or open up the menu on the top right corner of the page. Click on 'Profile' and below your avatar, click on the 'Edit' link. All information except for your email is public.
      </p>
      <div
        class="my-4 flex w-full justify-end"
      >
        <a
          href="/#top"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up-long "
            data-icon="arrow-up-long"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 9.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 109.3V480c0 17.7 14.3 32 32 32s32-14.3 32-32V109.3l73.4 73.4c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-128-128z"
              fill="currentColor"
            />
          </svg>
          <span
            class="text-mobile-label-1 ms-1"
          >
            Top
          </span>
        </a>
      </div>
      <h2
        class="pt-4 max-w-prose me-auto text-mobile-section-header font-medium my-4"
        id="link-studyBuddy"
      >
        Becoming and delisting from study buddy
      </h2>
      <p
        class="max-w-prose mx-auto"
      >
        By default, you are not listed as a study buddy on Pebbles. To list yourself so others can find you on the Study Buddy Board, go to edit your profile, click on edit study buddy, and toggle the 'Join study buddy' on. To unlist, simply toggle it off and the information you input for study buddy will be hidden from public.
      </p>
      <div
        class="my-4 flex w-full justify-end"
      >
        <a
          href="/#top"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up-long "
            data-icon="arrow-up-long"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 9.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 109.3V480c0 17.7 14.3 32 32 32s32-14.3 32-32V109.3l73.4 73.4c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-128-128z"
              fill="currentColor"
            />
          </svg>
          <span
            class="text-mobile-label-1 ms-1"
          >
            Top
          </span>
        </a>
      </div>
      <h2
        class="pt-4 max-w-prose me-auto text-mobile-section-header font-medium my-4"
        id="link-studyBuddyUse"
      >
        Using the Study Buddy Board
      </h2>
      <p
        class="max-w-prose mx-auto"
      >
        The board displays users in order of recently listed study buddies. To search using username, name, or their bio, type into the search bar and press Enter. To filter from categories, press the Filter button and choose the categories that you want to search for.
      </p>
      <div
        class="my-4 flex w-full justify-end"
      >
        <a
          href="/#top"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up-long "
            data-icon="arrow-up-long"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 9.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 109.3V480c0 17.7 14.3 32 32 32s32-14.3 32-32V109.3l73.4 73.4c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-128-128z"
              fill="currentColor"
            />
          </svg>
          <span
            class="text-mobile-label-1 ms-1"
          >
            Top
          </span>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
